core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -58,6 +58,20 @@ case class PSBT(
       vec.head.version
     }
   }
+
+  def isFinalized: Boolean = inputMaps.forall(_.isFinalized)
+
+  def combinePSBT(other: PSBT): PSBT = {
+    require(other.transaction.txId == this.transaction.txId,
+            "Can only combine PSBTs with the same global transaction.")
+
+    val global = globalMap.combine(other.globalMap)
+    val inputs = inputMaps.zipWithIndex.map(input =>
```

Comment:
rather than using `zipWithIndex` when you have two collections of the same size, you can actually just call `inputMaps.zip(other.inputMaps)` and get a Vector where they are paired by place/index